app/controllers/child_objects_controller.rb

```diff
@@ -66,6 +66,18 @@ def destroy
     end
   end
 
+  def update_checksum
+    respond_to do |format|
+      if @child_object.update(sha512_checksum: @child_object&.access_sha512_checksum)
```

Comment:
You could update the filesize here too. It can help with the integrity check and it might be good to have all the sizes.

I wonder if it should clear out the other checksums too, since this should replace them if they are set.